### ISSUE TITLE
DiffOpt accepts SingleVariable in ScalarSet but does not retrieve it

### ISSUE THREAD

**Issue by matbesancon** [jump-dev/DiffOpt.jl#67]
This is due to `get_problem_data(model::MOI.AbstractOptimizer)` fetching only ScalarAffineFunction in set, so SingleVariable is passed but not counted

**Comment by matbesancon** [comment]
Handling single variables is tricky, one part of the issue is that we don't check lower bounds here:
```julia
ERROR: MathOptInterface.LowerBoundAlreadySet{MathOptInterface.GreaterThan{Float64}, MathOptInterface.GreaterThan{Float64}}: Cannot add `SingleVariable`-in`MathOptInterface.GreaterThan{Float64}` constraint for variable MathOptInterface.VariableIndex(3) as a `SingleVariable`-in`MathOptInterface.GreaterThan{Float64}` constraint was already set for this variable and both constraints set a lower bound.
```

is thrown in certain cases, but not all.

**Comment by matbesancon** [comment]
There seems to be a more general issue that not all constraints are retrieved, only `SAF in LessThan` which ignores for example vectorized `Ax + b = 0`, has this been looked at? @AKS1996 @blegat

**Comment by matbesancon** [comment]
Currently we offer blanket support for some sets and functions which should probably not be there:
```
const SUPPORTED_SCALAR_SETS = Union{
    MOI.GreaterThan{Float64},
    MOI.LessThan{Float64},
    MOI.EqualTo{Float64},
    MOI.Interval{Float64}
}

const SUPPORTED_SCALAR_FUNCTIONS = Union{
    MOI.SingleVariable,
    MOI.ScalarAffineFunction{Float64}
}

const SUPPORTED_VECTOR_FUNCTIONS = Union{
    MOI.VectorOfVariables,
    MOI.VectorAffineFunction{Float64},
}

const SUPPORTED_VECTOR_SETS = Union{
    MOI.Zeros,
    MOI.Nonpositives,
    MOI.Nonnegatives,
    MOI.SecondOrderCone,
    MOI.PositiveSemidefiniteConeTriangle,
}
```

**Comment by AKS1996** [comment]
> There seems to be a more general issue that not all constraints are retrieved, only `SAF in LessThan` which ignores for example vectorized `Ax + b = 0`, has this been looked at? @AKS1996 @blegat

nope. `get_problem_data` isn't extracting all constraints - such as greater than con. this issue links to #14 , which i totally forgot

**Comment by matbesancon** [comment]
@be-apt I think this one can be closed?

**Comment by be-apt** [comment]
yep, was fixed in #77. Diffopt retrieves SingleVariable in <, > constraints
